### DISCUSSION TITLE
Use more standard method for registering GF with Thor

### DISCUSSION THREAD

**Discussion by Jimrnelson** [VFPX/GoFish#149] [General]
The standard way for registering GF with Thor (supported by VFPX Deployment) is that the creation of the Thor Tool be created within the Version file.

GF actually has this embedded in the GoFish.App.  

(For reasons I could never grasp, Matt Slay wanted the Thor Tools to be registered every time GF was called, which meant it had to be baked into GF.)

**Comment by lscheffler** [comment]
I vote for removing this from GF.

**Comment by Jimrnelson** [comment]
@lscheffler 

> I vote for removing this from GF.

If you mean by this removing the code from GoFish.App and adding it instead to the Version file (which only exists and is referenced as part of Thor CFU), I heartedly agree.

**Comment by lscheffler** [comment]
@Jimrnelson 
Hi Jim, may I ask your support on this?  
I'm moving forward, but I have no good idea about the Thor stuff to enter.   

The most recent development state is at https://github.com/lscheffler/GoFish, clone git@github.com:lscheffler/GoFish.git and use the dev branch. The file that should hold the Thor code is `BuildProcess\Thor_Update_GoFish.prg` (https://github.com/lscheffler/GoFish/blob/dev/BuildProcess/Thor_Update_GoFish.prg)

Could you alter this so that it does what it should do?

Thanks

**Comment by Jimrnelson** [comment]
Sure, will do so over the weekend if time permits.

This is mostly migrating code from with GF.App into the version file.  However, it's tricky to test.

**Reply by lscheffler** [reply]
With some luck - and some patience on your side - we can go through this together, so you get an example for simple branching - and what I mean with independent testing. I prepare this anyway for my tests.

**Reply by Jimrnelson** [reply]
If you want to use this as a way to introduce me to branching, it will have to wait until well into January.

I actually don't need any of your updated source code to provide a solution to this issue, as I will only be modifying your version template file (which I can attach here).

**Reply by lscheffler** [reply]
Okay.
But it would be really simple, just to enable the testing. Nothing fancy. No harm to existing stuff. 3 commands on git you enter blindly, the rest as you do all the time. VFPXDeployment. And set up a vanilla Thor, copy a file to the *My Updates* folder, then run CFU.

**Reply by Jimrnelson** [reply]
OK, then, this can certainly wait until at least late January.  

That's actually better for me, as I was gonna have to find the time to squeeze this in if I worked on it over the weekend.

I've got plenty going on to keep me busy until then.

**Reply by lscheffler** [reply]
In this case, just send me the file, I do the testing.

Edit: ... and we keep it outside git history.

**Reply by Jimrnelson** [reply]
OK, will do.

**Comment by Jimrnelson** [comment]
This has been taken care of.  Months ago, actually.
